src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -894,9 +903,32 @@ def read_file(
                     )
                 )
             )
+
+            def _get_unaliased_name(unaliased: ColumnOrName):
+                if isinstance(unaliased, Column):
+                    if isinstance(unaliased._expression, Alias):
+                        return unaliased._expression.child.sql
+                    return unaliased.get_name()
+                return unaliased
+
+            try:
+                combined_schema = [
+                    Attribute(
+                        metadata_col.get_name(),
+                        METADATA_COLUMN_TYPES[
+                            _get_unaliased_name(metadata_col).upper()
+                        ],
+                    )
+                    for metadata_col in metadata_columns
+                ] + schema
```

Comment:
```suggestion
                    for metadata_col in metadata_columns or []
                ] + schema
```

Reply:
Is could be why some tests are failing

Reply:
Thanks! I also realized the same after observing the test failure lol